networklibrary/src/main/java/com/eis/smsnetwork/SMSNetDictionary.java

```diff
@@ -46,33 +53,120 @@ public void removeResource(String key) {
      * @param key The key which defines the resource to get
      * @return Returns a resource corresponding to the key if present in the dictionary,
      * else returns null
-     * @throws IllegalArgumentException If the key is invalid.
-     *                                  A key is said to be valid only if it's composed of a single word
+     * @throws IllegalArgumentException if the argument contains a backslash as its last
+     *                                  character.
      */
     public String getResource(String key) {
         checkKeyValidity(key);
         return dict.get(key);
     }
 
     /**
-     * Checks if a given key is valid, else throws IllegalArgumentException.
-     * A key is said to be valid only if it's composed of one word.
+     * Removes all keys and resources from the dictionary.
+     */
+    public void clear() {
+        dict.clear();
+    }
+
+    /**
+     * Adds a resource parsed from an SMS to the network dictionary
+     *
+     * @param key      The key which defines the resource
+     * @param resource The resource to add
+     * @throws IllegalArgumentException if one of the arguments contains a backslash as its last
+     *                                  character.
+     */
+    public void addResourceFromSMS(@NonNull String key, @NonNull String resource) {
```

Comment:
maybe you need those methods cause other choices in structure
but I think that if a user can access removeEscapes he can use the above set of methods

Reply:
after I saw escapes methods are public maybe there's a point in make a class that encapsulates these "translations"

Reply:
The problem is, even if I put those methods in a class, I still have to make those functions public, because they're used in different packages

Reply:
you can mark this as resolved